internal/compiler/program.go

```diff
@@ -1031,7 +1031,10 @@ func (p *Program) GetGlobalDiagnostics(ctx context.Context) []*ast.Diagnostic {
 		return nil
 	}
 
-	pool := p.checkerPool.(*checkerPool)
+	pool, ok := p.checkerPool.(*checkerPool)
+	if !ok {
+		return nil // !!! Global diagnostics in the editor?
```

Comment:
Whenever we get around to revamping how the checkers are allocated, I think the fix is going to be to (in the editor) collect global diags from diagnostic-producing checkers, and then re-collect them whenever we check a new file, then expose those diags there.